mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,442 @@
+import gg.solarmc.loader.schema.tables.records.ClansClanEnemiesRecord;
+import gg.solarmc.loader.schema.tables.records.ClansClanInfoRecord;
+import gg.solarmc.loader.schema.tables.records.ClansClanMembershipRecord;
+import org.jooq.DSLContext;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static gg.solarmc.loader.schema.tables.ClansClanAlliances.CLANS_CLAN_ALLIANCES;
+import static gg.solarmc.loader.schema.tables.ClansClanEnemies.CLANS_CLAN_ENEMIES;
+import static gg.solarmc.loader.schema.tables.ClansClanMembership.CLANS_CLAN_MEMBERSHIP;
+import static gg.solarmc.loader.schema.tables.ClansClanInfo.CLANS_CLAN_INFO;
+
+/**
+ * Note to reader of this source code: Come play SolarMC!
+ */
+@SuppressWarnings("unused")
+public class Clan {
+
+    private final int clanID;
+    private final String clanName;
+
+    private Set<ClanMember> members;
+    private final ClanManager manager;
+    private final ClanMember leader;
+
+    private volatile int clanKills;
+    private volatile int clanDeaths;
+    private volatile int clanAssists;
+
+    public Clan(int clanID, String clanName, int clanKills, int clanDeaths, int clanAssists,
+                ClanManager manager, Set<ClanMember> members, ClanMember leader) {
+
+        this.clanID = clanID;
+        this.clanName = clanName;
+        this.clanKills = clanKills;
+        this.clanDeaths = clanDeaths;
+        this.clanAssists = clanAssists;
+        this.members = Set.copyOf(members);
+        this.manager = manager;
+        this.leader = leader;
+    }
+
+    public int getID() {
+        return this.clanID;
+    }
+
+    public String getName() {
+        return this.clanName;
+    }
+
+    /**
+     * Returns the kills at the time of querying (not accurate)
+     * @return kills
+     */
+    public int currentKills() {
+        return clanKills;
+    }
+
+    /**
+     * Returns the deaths at the time of querying (not accurate)
+     * @return deaths
+     */
+    public int currentDeaths() {
+        return clanDeaths;
+    }
+
+    /**
+     * Returns the ass at the time of querying (not accurate)
+     * @return ass
+     */
+    public int currentAssists() {
+        return clanAssists;
+    }
+
+    /**
+     * Gets all current members currently in the clan. Not accurate.
+     * @return all ClanMembers.
+     */
+    public Set<ClanMember> currentMembers() {
+        return this.members;
+    }
+
+    /**
+     * Returns a number representing the currently allied clan by cache.
+     * @return currently allied clan ID.
+     */
+    public Optional<Integer> currentlyAlliedClanID() {
+        return manager.getAllyFromCache(this.getID());
+    }
+
+    /**
+     * Gets this object's currently allied clan accurately
+     * @param transaction the tx
+     * @return optional holding currently allied clan
+     */
+    public Optional<Clan> getAlliedClan(Transaction transaction) {
+        ClansClanAlliancesRecord rec1 = transaction
```

Comment:
It is okay that you are using jooq's record instead of traditional SQL. However, I recommend you avoid selecting all columns going forward, provided it makes and the record does not add much value. See https://blog.jooq.org/2017/03/08/many-sql-performance-problems-stem-from-unnecessary-mandatory-work/